kujin-CSharp/Heap/heap_p2.cs

```diff
@@ -0,0 +1,202 @@
+                { 35, 43 }, { 26, 1 }, { 100, 1 }, { 101, 100 }, { 10000, 17 }, { 0, 10000 },
+                { 0, 1000 }, { 563, 51 }, { 986, 761 }, { 21, 16 }, { 65, 1 }, { 100, 311 }, { 0, 1 }, { 0, 1 },
+                { 0, 1 }, { 0, 1 },
+                { 0, 1 }, { 0, 1 }, { 0, 1 }, { 2000, 0 }, { 20, 1 }, { 0, 1 }, { 6, 1 }, { 0, 112 }, { 0, 1 }, { 0, 1 }
+            };
+
+
+          
+
+             Assert.True(solution(input_1) == 7);
+            Assert.True(solution(input_2) == 15);
+            Assert.True(solution(input_3) == 74);
+             Assert.True(solution(input_3) == solution(input_3_1));
+             Assert.True(solution(input_4) == 3);
+             Assert.True(solution(input_5) == 10);
+             Assert.True(solution(input_6) == 13);
+             Assert.True(solution(input_7) == 5);
+             Assert.True(solution(input_8) == 3);
+             Assert.True(solution(input_9) == 14);
+             Assert.True(solution(input_10) == 25);
+            Assert.True(solution(input_11) == 10);
+            Assert.True(solution(input_12) == 6);
+            Assert.True(solution(input_13) == 7);
+            Assert.True(solution(input_14) == 15);
+            Assert.True(solution(input_15) == 72);
+            Assert.True(solution(input_16) == 3);
+            Assert.True(solution(input_17) == 5008);
+            Assert.True(solution(input_18) == 1);
+            Assert.True(solution(input_19) == 50);
+            Assert.True(solution(input_20) == 12);
+            Assert.True(solution(input_21) == 9);
+            Assert.True(solution(input_22) == 2);
+            Assert.True(solution(input_23) == 6);
+            Assert.True(solution(input_24) == 19);
+            Assert.True(solution(input_25) == 50);
+            Assert.True(solution(input_26) == 7);
+            Assert.True(solution(input_27) == 0);
+            Assert.True(solution(input_28) == 100);
+            Assert.True(solution(input_29) == 10);
+            Assert.True(solution(input_30) == 60);
+            //
+            //
+
+        }
+
+        public int solution(int[,] jobs)
+        {
+            //시간 값
+            int time = 0;
+            int arrayIndex = 0;
+
+            KeyValuePair<int, int> temp;
+            List<KeyValuePair<int, int>> insertJobList = new List<KeyValuePair<int, int>>();
+            List<KeyValuePair<int, int>> jobList = new List<KeyValuePair<int, int>>();
+            List<int> timeList = new List<int>();
+
+
+            for (int i = 0; i < jobs.GetLength(0); i++)
+            {
+                insertJobList.Add(new KeyValuePair<int, int>(jobs[i, 0], jobs[i, 1]));
+            }
+
+            //시간순으로 정렬
+            insertJobList = insertJobList.OrderBy(x => x.Key).ToList();
+
+
+            while (jobs.GetLength(0) > arrayIndex || jobList.Count != 0)
+            {
+                if (jobList.Count == 0 && jobs[arrayIndex, 0] > time)
```

Comment:
```suggestion
                if (jobList.Count == 0 && insertJobList[arrayIndex, 0] > time)
```
문맥적으로 `jobs`보단 `insertJobList`가 들어와야될 것 같음.

Reply:
이게 문제였나봄;; 감사감사